com.unity.netcode.gameobjects/Runtime/NetworkVariable/CollectionSerializationUtility.cs

```diff
@@ -0,0 +1,664 @@
+using System;
+using System.Collections.Generic;
+using Unity.Collections;
+using Unity.Collections.LowLevel.Unsafe;
+using Unity.Mathematics;
+
+namespace Unity.Netcode
+{
+    internal static class CollectionSerializationUtility
+    {
+        public static void WriteNativeArrayDelta<T>(FastBufferWriter writer, ref NativeArray<T> value, ref NativeArray<T> previousValue) where T : unmanaged
+        {
+            using var changes = new ResizableBitVector(Allocator.Temp);
+            int minLength = math.min(value.Length, previousValue.Length);
+            var numChanges = 0;
+            for (var i = 0; i < minLength; ++i)
+            {
+                var val = value[i];
+                var prevVal = previousValue[i];
+                if (!NetworkVariableSerialization<T>.AreEqual(ref val, ref prevVal))
+                {
+                    ++numChanges;
+                    changes.Set(i);
+                }
+            }
+
+            for (var i = previousValue.Length; i < value.Length; ++i)
+            {
+                ++numChanges;
+                changes.Set(i);
+            }
+
+            if (changes.GetSerializedSize() + FastBufferWriter.GetWriteSize<T>() * numChanges > FastBufferWriter.GetWriteSize<T>() * value.Length)
+            {
+                writer.WriteByteSafe(1);
+                writer.WriteValueSafe(value);
+                return;
+            }
+            writer.WriteByte(0);
+            BytePacker.WriteValuePacked(writer, value.Length);
+            writer.WriteValueSafe(changes);
+            unsafe
+            {
+                var ptr = (T*)value.GetUnsafePtr();
+                var prevPtr = (T*)previousValue.GetUnsafePtr();
+                for (int i = 0; i < value.Length; ++i)
+                {
+                    if (changes.IsSet(i))
+                    {
+                        if (i < previousValue.Length)
+                        {
+                            NetworkVariableSerialization<T>.WriteDelta(writer, ref ptr[i], ref prevPtr[i]);
+                        }
+                        else
+                        {
+                            NetworkVariableSerialization<T>.Write(writer, ref ptr[i]);
+                        }
+                    }
+                }
+            }
+        }
+        public static void ReadNativeArrayDelta<T>(FastBufferReader reader, ref NativeArray<T> value) where T : unmanaged
+        {
+            reader.ReadByteSafe(out byte full);
+            if (full == 1)
```

Comment:
Another place, like a brief comment of what the full byte means like:
"If this is a full update, dispose of the current and read in the full value".